### PR TITLE
IGVF-118-deploy-frontend-with-cdk

### PR DESCRIPTION
**Pull request by keenangraham**

### CODE REVIEW

File: libs/constants.js [hunk 1]
```diff
@@ -1,6 +1,12 @@
+import getConfig from 'next/config'
+
+
+const { serverRuntimeConfig, publicRuntimeConfig } = getConfig()
+
+
```
Comment (forresttanaka) [inline libs/constants.js:6]: For this and line 3, we have ESLint configured to error for two consecutive blank lines.

File: pages/index.js [hunk 2]
```diff
@@ -19,3 +19,9 @@ const Home = () => {
 }
 
 export default Home
+
+export const getServerSideProps = async () => {
+  return {
+      props: {},
```
Comment (forresttanaka) [inline pages/index.js:25]: Need to indent two spaces less to pass prettier.

File: next.config.js [hunk 3]
```diff
@@ -7,4 +7,10 @@ module.exports = {
       "s.gravatar.com",
     ],
   },
+  serverRuntimeConfig: {
+    BACKEND_URL: process.env.BACKEND_URL,
+  },
+  publicRuntimeConfig: {
+    PUBLIC_BACKEND_URL: process.env.PUBLIC_BACKEND_URL || process.env.BACKEND_URL,
```
Comment (forresttanaka) [inline next.config.js:14]: Need to resave this file with prettier; it wants to break this line.

File: jest.config.js [hunk 4]
```diff
@@ -9,6 +9,9 @@ const customJestConfig = {
     "<rootDir>/components",
     "<rootDir>/libs",
   ],
+  modulePathIgnorePatterns: [
+    "<rootDir>/cdk/"
+  ],
```
Comment (forresttanaka) [inline jest.config.js:14]: Prettier wants no line breaks for line 12.

**Review by forresttanaka — CHANGES_REQUESTED** [review]
Looks great! Just have to resave a couple of files with prettier installed, and one file runs afoul of ESLint.

File: cdk/infrastructure/constructs/existing/igvf_dev.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
+from constructs import Construct
+
+from shared_infrastructure.igvf_dev.connection import CodeStarConnection
```
Comment (hitz) [inline cdk/infrastructure/constructs/existing/igvf_dev.py:3]: where is this module, I assume it's what links igvfd CDK to igvf-ui?

File: docker/nextjs/Dockerfile [hunk 6]
```diff
@@ -1,5 +1,7 @@
 FROM node:16.14.0-alpine
 
+ENV NEXT_TELEMETRY_DISABLED=1
```
Comment (hitz) [inline docker/nextjs/Dockerfile:3]: Is there a specific technical reason for disabling telemetry, or are we just being private?

**Review by hitz — APPROVED** [review]
Looks fine to me, just some random questions about things, nothing breaking.

**Review by forresttanaka — APPROVED** [review]

File: cdk/README.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+ * `cdk diff`        compare deployed stack with current state
+ * `cdk docs`        open CDK documentation
+
+## Run type checking with mypy
+```
+# In cdk folder.
+$ pip install -r requirements.txt -r requirements-dev.txt
+$ mypy .
+```
+Runs in strict mode, excluding `test` folder.
```
[the remark this answers is not in the document]
Comment (keenangraham) [reply cdk/README.md:105]: Yes, CDK code is in Python. We could write same patterns in Typescript, but the `shared_infrastructure` code is in Python, so easier to share if we use Python here.

File: cdk/infrastructure/constructs/existing/igvf_dev.py (hunk 5 above)
Comment (keenangraham) [reply cdk/infrastructure/constructs/existing/igvf_dev.py:3]: This is its own repo that allows you to use things that already exist within an AWS account (such as VPCs and Slack connections): https://github.com/igvf-dacc/shared-infrastructure

We're not actually deploying anything here, just making existing things importable, so you don't have to redefined things (e.g. the account environment) for every project.

File: docker/nextjs/Dockerfile (hunk 6 above)
Comment (keenangraham) [reply docker/nextjs/Dockerfile:3]: Figure we don't need overhead of sending tracking data in background in production.

**Pull request merged**